src/Refactoring-Core/RBPullUpMethodRefactoring.class.st

```diff
@@ -340,23 +400,22 @@ RBPullUpMethodRefactoring >> removeDuplicateMethods [
 
 { #category : 'transforming' }
 RBPullUpMethodRefactoring >> removeDuplicatesOf: aSelector [
+
 	| tree |
 	tree := targetSuperclass parseTreeForSelector: aSelector.
-	targetSuperclass allSubclasses do:
-			[:each |
-			((each directlyDefinesMethod: aSelector) and:
-					[(tree equalTo: (each parseTreeForSelector: aSelector) exceptForVariables: #())
-						and: [(each superclass whoDefinesMethod: aSelector) == targetSuperclass]])
-				ifTrue:
-					[removeDuplicates
-						ifFalse:
-							[removeDuplicates :=
-							self
-								refactoringConfirmWarning: 'Do you want to remove duplicate subclass methods?'].
-					removeDuplicates ifTrue:[
-						self generateChangesFor:
-							(RBRemoveMethodTransformation
-								selector: aSelector from: each)]]]
+	targetSuperclass allSubclasses do: [ :each |
+		((each directlyDefinesMethod: aSelector) and: [
+			 (tree
+				  equalTo: (each parseTreeForSelector: aSelector)
+				  exceptForVariables: #(  )) and: [
+				 (each superclass whichClassIncludesSelector: aSelector)
+				 == targetSuperclass ] ]) ifTrue: [
+			removeDuplicates ifFalse: [
+				removeDuplicates := self refactoringConfirmWarning:
+					                    'Do you want to remove duplicate subclass methods?' ].
+			removeDuplicates ifTrue: [
+				self generateChangesFor:
+					(RBRemoveMethodTransformation selector: aSelector from: each) ] ] ]
```

Comment:
You are right, I created a new condition called `ReMethodsHaveNoDuplicatesCondition` that encapsulates this functionality

Reply:
I like the suggestion too.

Reply:
But this is not just suggestion, this is a cleanup action that will remove methods with duplicate body in all siblings. So you may need a condition, but you also need to be able to call this transformation as well. This can be done along with "Push up referenced instance variables"?

Reply:
@balsa-sarenac you are right that this is a cleanup action, but unlike "push up referenced instance variables", this action is not fixing a breaking change. In fact, now that I think about it better, I hesitate to put `ReMethodsHaveNoDuplicatesCondition` together with other breaking change conditions, since this condition is not breaking the refactoring behavior

Reply:
For the moment I have implemented this cleanup action in a similar way as "push up referenced instance variables" (this is, using a breaking change condition together with a choice to apply the fix). But I think we should consider treating cleanup actions in a different way, because for now their "choices" appear in the dialog "there are potential breaking changes" as shown in the image 
<img width="927" alt="image" src="https://github.com/pharo-project/pharo/assets/4822303/ff768605-8b97-45a6-8164-05da4e6d74c5">